modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/ParameterGroup.java

```diff
@@ -46,10 +48,10 @@
     private final Class<?> type;
 
     /**
-     * The {@link Field} in which the generated value of
-     * {@link #type} is to be assigned
+     * The {@link Field} in which the generated value of {@link #type} is to be assigned.
+     * If the {@link ParameterGroup} is used as an argument of an operation this fill will be {@link Optional#empty()}
      */
-    private final Field field;
+    private final Optional<Field> field;
```

Comment:
this is a hack. You're just using Optional.empty() as a flag. Can't you just have two implementations?